src/js/comments.js

```diff
@@ -16,38 +17,50 @@ class Comments {
 	}
 
 	/**
-	 * Render comment instances and store them on the `_comments` property.
+	 * Render a comments instance authenticated with Coral Talk.
 	 *
 	 * @access private
-	 * @returns {void}
+	 * @returns {HTMLElement}
 	 */
-	_renderComments() {
+	_renderComments () {
 		/*global Coral*/
-		const scriptElement = document.createElement('script');
-		scriptElement.src = 'https://ft-next-talk-spike.herokuapp.com/static/embed.js';
-		scriptElement.onload = () => Coral.Talk.render(document.querySelector('[data-o-component="o-comments"]'),
-			{
-				talk: 'https://ft-next-talk-spike.herokuapp.com',
-				events: (events) => {
-					events.onAny((name, data) => {
-						const message = new CustomEvent('talkEvent', {
-							detail: {
-								name,
-								data
-							}
-						});
-						parent.dispatchEvent(message);
-					});
-				}
-			}
-		);
-
-		this.oCommentsEl.appendChild(scriptElement);
+		fetch(`https://comments-auth.ft.com/v1/jwt/`, {
+			credentials: 'include'
+		}).then(res => res.json())
+			.then(json => json.token)
+			.then(token => {
+				const scriptElement = document.createElement('script');
+				scriptElement.src = 'https://ft-next-talk-spike.herokuapp.com/static/embed.js';
+				scriptElement.onload = () => Coral.Talk.render(document.querySelector('[data-o-component="o-comments"]'),
+					{
+						talk: 'https://ft-next-talk-spike.herokuapp.com',
+						auth_token: token,
+						events: (events) => {
+							events.onAny((name, data) => {
+								const message = new CustomEvent('talkEvent', {
+									detail: {
+										name,
+										data
+									}
+								});
+								parent.dispatchEvent(message);
+							});
+						}
+					}
+				);
+				this.oCommentsEl.appendChild(scriptElement);
+				/**
+				 * In order to test the asynchronous function, send an event when fetch returns and
+				 * the script element is injected into the document.
+				 */
+				document.dispatchEvent(new Event('oCommentsReady'));
```

Comment:
@GlynnPhillips would `widget.ready` be similar to `oCommentsReady`?

Reply:
@ker-an Interesting I hadn't noticed this event in here. I think we would end up removing this and replacing it with one of ours. I think they would be similar though yes

Reply:
It may depend on when `widget.ready` event is fired. We needed this event because we do asynchronous initialization here, and we would like to notify the unit tests when it is complete.